app/javascript/components/rooms/room/Room.jsx

```diff
@@ -42,8 +43,19 @@ export default function Room() {
   const startMeeting = useStartMeeting(friendlyId);
   const currentUser = useAuth();
   const localizedTime = localizeDayDateTimeString(room?.last_session, currentUser?.language);
+  const roomSettings = useRoomSettings(friendlyId);
 
-  function copyInvite() {
+  function copyInvite(role) {
+    if (role === 'viewer') {
+      navigator.clipboard.writeText(roomSettings?.data?.glViewerAccessCode);
+      toast.success(t('toast.success.room.copied_viewer_code'));
+      return;
+    }
+    if (role === 'moderator') {
+      navigator.clipboard.writeText(roomSettings?.data?.glModeratorAccessCode);
+      toast.success(t('toast.success.room.copied_moderator_code'));
+      return;
+    }
     navigator.clipboard.writeText(`${window.location}/join`);
     toast.success(t('toast.success.room.copied_meeting_url'));
```

Comment:
Why not just do an `if else if` here? Rather than add the if + return

Reply:
fixed!